File: iOS/Code/Swift/bugTrap/SampleHostApp/Controllers/AppsCollectionViewCell.swift
//
//  AppsCollectionViewCell.swift
//  SampleHostApp
//
//  Created by Colby Williams on 8/22/15.
//  Copyright © 2015 bugTrap. All rights reserved.
//

import UIKit;
import CoreGraphics;

class AppsCollectionViewCell : UICollectionViewCell
{
	@IBOutlet weak var BackgroundStyleView: UIView!
	
	@IBOutlet weak var AppIconImageView: UIImageView!
	
	@IBOutlet weak var AppNameValueLabel: UILabel!
	
	@IBOutlet weak var SessionsInLastValueLabel: UILabel!
	
	@IBOutlet weak var SessionsInLastLabel: UILabel!
	
	@IBOutlet weak var OpenIssuesValueLabel: UILabel!
	
	@IBOutlet weak var OpenIssuesLabel: UILabel!
	
	@IBOutlet weak var UsersInLastValueLabel: UILabel!
	
	@IBOutlet weak var UsersInLastLabel: UILabel!
	
	@IBOutlet weak var UsersExperiencingIssuesValueLabel: UILabel!
	
	@IBOutlet weak var UsersExperiencingIssuesLabel: UILabel!
	
	
	var configuredShadow = false
	
	func setContent(cellSize: CGSize, app: String) {
		
		AppNameValueLabel.text = app;
		
		if !configuredShadow {
			
			configuredShadow = true
			
			BackgroundStyleView.layer.shadowOffset = CGSize.zeroSize;
			BackgroundStyleView.layer.shadowColor = UIColor.blackColor().CGColor;
			BackgroundStyleView.layer.cornerRadius = 3;
			BackgroundStyleView.layer.shadowRadius = 4;
			BackgroundStyleView.layer.shadowOpacity = 0.1;
			BackgroundStyleView.layer.shadowPath = UIBezierPath(rect: CGRect (x: 0, y: 4, width: cellSize.width - 12, height: cellSize.height - 14)).CGPath;
		}
	}
}

File: iOS/Code/Xamarin/Insights/Insights.iOS/ViewControllers/AppsCollectionViewCell.cs
using System;
using UIKit;
using CoreGraphics;

namespace Insights.iOS
{
	public partial class AppsCollectionViewCell : UICollectionViewCell
	{
		public static readonly string ReuseId = "AppsCollectionViewCell";

		bool configuredShadow;

		public AppsCollectionViewCell (IntPtr handle) : base(handle)
		{
			
		}


		public void SetContent (CGSize cellSize, MobileApp app)
		{
			AppNameValueLabel.Text = app.Name;

			if (!configuredShadow) {
				configuredShadow = true;

				BackgroundStyleView.Layer.ShadowOffset = CGSize.Empty;
				BackgroundStyleView.Layer.ShadowColor = UIColor.Black.CGColor;
				BackgroundStyleView.Layer.CornerRadius = 3f;
				BackgroundStyleView.Layer.ShadowRadius = 4f;
				BackgroundStyleView.Layer.ShadowOpacity = 0.1f;
				BackgroundStyleView.Layer.ShadowPath = UIBezierPath.FromRect(new CGRect (0, 4, cellSize.Width - 12, cellSize.Height - 14)).CGPath;
			}
		}
	}
}